ulauncher/modes/extensions/extension_runtime.py

```diff
@@ -79,6 +114,35 @@ def _kill(self) -> None:
     def read_stderr_line(self) -> None:
         self.error_stream.read_line_async(GLib.PRIORITY_DEFAULT, None, self.handle_stderr)
 
+    def send_message(self, message: dict[str, Any]) -> None:
+        """Send a JSON message to extension."""
+        try:
+            json_str = json.dumps(message)
+            self.msg_socket.write(json_str)
+            logger.debug("Sent message to %s: %s", self.ext_id, message)
+        except Exception:
+            logger.exception("Failed to send message to %s", self.ext_id)
```

Comment:
Should we close a connection or take some kind of action in these cases? Does it seem like it could actually be recoverable, such that the next message could be successful?

Reply:
I reworked the patch in 8b5788d4 now